temporal-serviceclient/src/main/java/io/temporal/serviceclient/WorkflowServiceStubsOptions.java

```diff
@@ -461,7 +461,16 @@ public Builder setTarget(String target) {
       return this;
     }
 
-    /** Sets the rpc timeout value for non-query and non-long-poll calls. Default is 10 seconds. */
+    /**
+     * Sets the rpc timeout value for non-query and non-long-poll calls. Default is 10 seconds.
+     *
+     * <p>This timeout is applied to a single rpc call happening under the hood of temporal client,
+     * but it's not an enforced timeout for a whole client-server interaction. In case of failure,
+     * the requests get automatically retried according to {@link
+     * #setRpcRetryOptions(RpcRetryOptions)}. The full timeout for an interaction is limited by
+     * {@link RpcRetryOptions.Builder#setExpiration(Duration)} and {@link
+     * RpcRetryOptions.Builder#setMaximumAttempts(int)}} whichever happens first.
```

Comment:
```suggestion
     * <p>This timeout is applied to only a single rpc call within a temporal client,
     * not a complete client-server interaction. In case of failure,
     * the requests are automatically retried according to {@link
     * #setRpcRetryOptions(RpcRetryOptions)}. The full timeout for an interaction is limited by
     * {@link RpcRetryOptions.Builder#setExpiration(Duration)} or {@link
     * RpcRetryOptions.Builder#setMaximumAttempts(int)}}, whichever happens first.
```